### PR TITLE
[Kubernetes] Add state namespace datastream

### PR DESCRIPTION
**Pull request by constanca-m**
## What

Add state namespace data stream.

The metricset was added in version 8.11.0.

Dimensions for TSDB:
- `kubernetes.namespace` is unique, so it is a dimension.
- `agent.id` is a common dimension among data streams.
- `orchestrator.cluster.url` is a dimension for all kubernetes data streams.

<details>

<summary>Results from running TSDB testing kit:</summary>

```
The time series fields for the TSDB index are: 
	- dimension (3 fields):
		- agent.id
		- kubernetes.namespace
		- orchestrator.cluster.url
	- gauge (1 fields):
		- kubernetes.state_namespace.created.sec
	- routing_path (3 fields):
		- agent.id
		- kubernetes.namespace
		- orchestrator.cluster.url

Index tsdb-index-enabled successfully created.

Copying documents from .ds-metrics-kubernetes.state_namespace-default-2023.10.30-000001 to tsdb-index-enabled...
All 210 documents taken from index .ds-metrics-kubernetes.state_namespace-default-2023.10.30-000001 were successfully placed to index tsdb-index-enabled.
```
</details>

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

1. Clone this branch.
2. Build the package.
3. Update the service registry.
4. Add Kubernetes to a policy and check the new data stream.

## Related issues

Relates to https://github.com/elastic/elastic-agent/issues/3100.

## Screenshots

In Discover:
image

In the integration itself:
image

### CODE REVIEW

File: packages/kubernetes/data_stream/state_namespace/fields/fields.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  type: group
+  description: >
+    Kubernetes namespace metrics.
+
+  fields:
+    - name: created.sec
+      type: double
+      unit: s
+      description: >
+        Unix creation timestamp.
```
Comment (gizas) [inline packages/kubernetes/data_stream/state_namespace/fields/fields.yml:11]: Can we alighn it with this https://github.com/elastic/integrations/blob/7750e0632f9d54b34a0ae289e1d3ff231f1eb166/packages/kubernetes/data_stream/state_cronjob/fields/fields.yml#L24C7-L25C61

So to add Epoch and also the metric type?

Comment (constanca-m) [reply packages/kubernetes/data_stream/state_namespace/fields/fields.yml]: Done!

**Comment by gizas** [comment]
Another idea is to consider if we need to add this info of namespace in our dashboards. To add it in Overview or in Pods?

@MichaelKatsoulis , @tetianakravchenko , @ChrsMark suggestions?

**Review by gizas — APPROVED** [review]

File: packages/kubernetes/data_stream/state_namespace/fields/agent.yml [hunk 2]
```diff
@@ -0,0 +1,198 @@
+- name: cloud
+  title: Cloud
+  group: 2
+  description: Fields related to the cloud or infrastructure the events are coming from.
+  footnote: 'Examples: If Metricbeat is running on an EC2 host and fetches data from its host, the cloud info contains the data about this machine. If Metricbeat runs on a remote machine outside the cloud and fetches data from a service running in the cloud, the field contains cloud data from the machine the service is running on.'
+  type: group
+  fields:
+    - name: account.id
```
Comment (tetianakravchenko) [inline packages/kubernetes/data_stream/state_namespace/fields/agent.yml:8]: we don't have any `dimension` for the info that identify the cloud provider, so in case there will be 2 clusters with the same name, same namespaces, and metrics will be stored in the same stack - I believe some documents will be dropped, isn't it?

File: packages/kubernetes/data_stream/state_namespace/fields/agent.yml [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+      level: extended
+      type: keyword
+      ignore_above: 1024
+      description: 'The cloud account or organization id used to identify different entities in a multi-tenant environment.
+
+        Examples: AWS account id, Google Cloud ORG Id, or other unique identifier.'
+      example: 666777888999
+    - name: availability_zone
```
Comment (tetianakravchenko) [inline packages/kubernetes/data_stream/state_namespace/fields/agent.yml:16]: also for such fields that are part of ecs we we can simply use `external: ecs` (similar to this PR https://github.com/elastic/integrations/pull/8100), instead of copping all field definition.
but it can be moved to another pr

File: packages/kubernetes/data_stream/state_namespace/fields/base-fields.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      description: >
+        Kubernetes deployment name
+
+    - name: statefulset.name
+      type: keyword
+      description: >
+        Kubernetes statefulset name
+
+    - name: job.name
+      dimension: true
```
Comment (tetianakravchenko) [inline packages/kubernetes/data_stream/state_namespace/fields/base-fields.yml:96]: why is this defined as a `dimension`?

File: packages/kubernetes/data_stream/state_namespace/fields/base-fields.yml [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      type: keyword
+      description: >
+        Kubernetes pod name
+
+    - name: pod.uid
+      type: keyword
+      description: >
+        Kubernetes pod UID
+
+    - name: pod.ip
```
Comment (tetianakravchenko) [inline packages/kubernetes/data_stream/state_namespace/fields/base-fields.yml:26]: do we actually need those fields that will be not available for the `state_namespace` datastream?

File: packages/kubernetes/data_stream/state_namespace/fields/fields.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+- name: kubernetes.state_namespace
+  type: group
+  description: >
+    Kubernetes namespace metrics.
+
+  fields:
+    - name: created.sec
```
Comment (tetianakravchenko) [inline packages/kubernetes/data_stream/state_namespace/fields/fields.yml:7]: should here be defined `metric_type` ?

File: packages/kubernetes/data_stream/state_namespace/fields/agent.yml (hunk 2 above)
Comment (constanca-m) [reply packages/kubernetes/data_stream/state_namespace/fields/agent.yml:8]: No, because we only have 1 metric, I consider this data stream to not be a candidate of TSDB. I will delete the other dimension, it was an accident. @tetianakravchenko

File: packages/kubernetes/data_stream/state_namespace/fields/base-fields.yml (hunk 4 above)
Comment (constanca-m) [reply packages/kubernetes/data_stream/state_namespace/fields/base-fields.yml]: It was a mistake, deleted it

File: packages/kubernetes/data_stream/state_namespace/fields/fields.yml (hunk 6 above)
Comment (constanca-m) [reply packages/kubernetes/data_stream/state_namespace/fields/fields.yml:7]: Since TSDB won't be enabled, then the metric type does not need to be defined.

File: packages/kubernetes/data_stream/state_namespace/fields/base-fields.yml (hunk 5 above)
Comment (constanca-m) [reply packages/kubernetes/data_stream/state_namespace/fields/base-fields.yml:26]: I copied from the other `state_*` data streams, but I will delete them, indeed. The only field that had data was `kubernetes.namespace`, so I will remove all the others. Thanks @tetianakravchenko

File: packages/kubernetes/data_stream/state_namespace/fields/agent.yml (hunk 3 above)
Comment (constanca-m) [reply packages/kubernetes/data_stream/state_namespace/fields/agent.yml:16]: Ok, since this is being done with all data streams, I will leave it as it is for now.

File: packages/kubernetes/data_stream/state_namespace/fields/fields.yml (hunk 6 above)
Comment (gizas) [reply packages/kubernetes/data_stream/state_namespace/fields/fields.yml:7]: Why would not be enabled?

Comment (constanca-m) [reply packages/kubernetes/data_stream/state_namespace/fields/fields.yml:7]: The advantages of TSDB are related to the metric types. This data stream only has one metric type, so there wouldn't be any. We followed a similar approach for other data streams that only had one metric, such as AWS Billing or GCP Billing @gizas

File: packages/kubernetes/manifest.yml [hunk 7]
```diff
@@ -1,15 +1,15 @@
...
+version: 1.52.0
 description: Collect logs and metrics from Kubernetes clusters with Elastic Agent.
 type: integration
 categories:
   - observability
   - containers
   - kubernetes
 conditions:
-  kibana.version: "^8.10.2"
+  kibana.version: "^8.11.0"
```
Comment (gizas) [inline packages/kubernetes/manifest.yml:12]: Also doubechecking is this 8.11.0 or 8.11.1?

Comment (constanca-m) [reply packages/kubernetes/manifest.yml:12]: It is 8.11.0

File: packages/kubernetes/data_stream/state_namespace/fields/fields.yml (hunk 6 above)
Comment (constanca-m) [reply packages/kubernetes/data_stream/state_namespace/fields/fields.yml:7]: After TSDB meeting discussion, we are enabling TSDB in all metric data streams after all. I updated the commits and will update the description now.

**Review by tetianakravchenko — APPROVED** [review]
